tools/sql_extraction/src/main/kotlin/Environment.kt

```diff
@@ -0,0 +1,96 @@
+package com.google.cloud.sqlecosystem.sqlextraction
+
+import com.google.cloud.sqlecosystem.sqlextraction.output.QueryFragment
+
+/**
+ * A mapping from variable name to all possible query fragment values
+ */
+class Environment {
+    private var parentScope: Environment?
+    private var reachingDefs: HashMap<String, QueryFragment?>
```

Comment:
The name `reachingDefs` is less than ideal. I would use something like `variableUsages` or `variableReferences`.